subsys/zigbee/cli/zigbee_cli_cmd_bdb.c

```diff
@@ -0,0 +1,748 @@
+ *    - `zr` it it is a router,
+ *    - `zed` if it is an end device.
+ *
+ * If the optional argument is provided, set the device role to `role`. Can be
+ * either `zc` or `zr`. <br>
+ *
+ * @note Zigbee End Device is not currently supported on the CLI Agent.
+ *
+ */
+static int cmd_zb_role(const struct shell *shell, size_t argc, char **argv)
+{
+	zb_nwk_device_type_t role;
+
+	if (argc == 1) {
+		role = zb_get_network_role();
+		if (role == ZB_NWK_DEVICE_TYPE_NONE) {
+			role = default_role;
+		}
+
+		if (role == ZB_NWK_DEVICE_TYPE_COORDINATOR) {
+			shell_print(shell, "zc");
+		} else if (role == ZB_NWK_DEVICE_TYPE_ROUTER) {
+			shell_print(shell, "zr");
+		} else if (role == ZB_NWK_DEVICE_TYPE_ED) {
+			shell_print(shell, "zed");
+		}
+	} else if (argc == 2) {
+		if (zb_cli_is_stack_started()) {
+			zb_cli_print_error(shell, "Stack already started",
+					   ZB_FALSE);
+			return -ENOEXEC;
+		}
+
+#ifdef ZB_ED_ROLE
+		if (!strcmp(argv[1], "zed")) {
+			default_role = ZB_NWK_DEVICE_TYPE_ED;
+			shell_print(shell, "End Device role set");
+		} else {
+			zb_cli_print_error(shell, "Role unsupported", ZB_FALSE);
+			return -ENOEXEC;
+		}
+#else
+		if (!strcmp(argv[1], "zc")) {
+			default_role = ZB_NWK_DEVICE_TYPE_COORDINATOR;
+			shell_print(shell, "Coordinator set");
+		} else if (!strcmp(argv[1], "zr")) {
+			default_role = ZB_NWK_DEVICE_TYPE_ROUTER;
+			shell_print(shell, "Router role set");
+		} else {
+			zb_cli_print_error(shell, "Role unsupported", ZB_FALSE);
+			return -ENOEXEC;
+		}
+#endif
+	}
+
+	zb_cli_print_done(shell, ZB_FALSE);
+	return 0;
+}
+
+/**@brief Start bdb top level commissioning process.
+ *
+ * This command can be called multiple times. If device is not on a network
+ * then it will attempt to join or form (depending on the role).
+ *
+ * If the device is on the network then the command will open the network for
+ * new devices to join.
+ *
+ * See Base Device Behaviour specification for details.
+ *
+ * @code
+ * > bdb start
+ * Started coordinator
+ * Done
+ * @endcode
+ */
+static int cmd_zb_start(const struct shell *shell, size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	uint32_t channel;
+	zb_bool_t ret = ZB_TRUE;
+	zb_uint8_t mode_mask = ZB_BDB_NETWORK_STEERING;
+
+	if ((!zb_cli_is_stack_started()) &&
+	    (zb_get_zboss_thread_id() == NULL)) {
+		channel = zb_get_bdb_primary_channel_set();
+
+		switch (default_role) {
+#ifndef ZB_ED_ROLE
+		case ZB_NWK_DEVICE_TYPE_ROUTER:
+			zb_set_network_router_role(channel);
+			shell_print(shell, "Started router");
+			break;
+
+		case ZB_NWK_DEVICE_TYPE_COORDINATOR:
+			zb_set_network_coordinator_role(channel);
+			shell_print(shell, "Started coordinator");
+			break;
+#else
+		case ZB_NWK_DEVICE_TYPE_ED:
+			zb_set_network_ed_role(channel);
+			shell_print(shell, "Started End Device");
+			break;
+#endif
+		default:
+			zb_cli_print_error(shell, "Role unsupported", ZB_FALSE);
+			return -ENOEXEC;
+		}
+
+		zigbee_enable();
```

Comment:
Should we indicate in the documentation that the stack (thread) is started/enabled in this case?

Reply:
Yes, it should but only command function documentation.As for the command help message it is fine as it is - with current implementation `bdb start` will start commissioning or create zboss thread and then start commissioning by itself.